### DISCUSSION TITLE
Tiempo de ejecución

### DISCUSSION THREAD

**Discussion by B4lladaresquintana** [IIC2133-PUC/2024-2#59] [Tarea 0] [answered]
Hola, en mi pc varía demasiado el tiempo a veces lo ejecuto y se tarda mas de 15 segundos, pero en otras todo bien. Traté de ejecutarlo en una instancia EC2 de AWS y se demoró menos de 0.5 segundos (El test hard-2). No se que podría hacer para que mi computador no influya considerablemente en el tiempo de ejecución.

**Comment by sebacarrascop** [comment, marked as answer]
hola!

Respecto a que se puede hacer con tu computador? por nuestra parte nada.

Sin embargo existen otras formas donde puede correr tu código en un entorno virtual como GitHub _Codespaces_.
Probé tu código en este ambiente  con el test hard-2 y obtuve los siguientes resultados:

<img width="763" alt="image" src="https://github.com/user-attachments/assets/b33b7adf-ba31-49ad-954a-0f451ad46f39">

Mientras que en el servidor del curso (donde se evaluará) se obtienen estos otros:

<img width="773" alt="image" src="https://github.com/user-attachments/assets/2885d010-f816-4c74-9d0e-c348a38b0bf9">

Igual hay una diferencia, pero no de magnitudes tan grandes como en tu computador llegando a correr en 15 segundos.
Te puedes guiar por tus resultados en _Codespaces_ como un límite superior para tus resultados en el servidor.
Con respecto a como usar _Codespaces_, tan solo es meterse al repositorio y apretar este botón:

![Screenshot 2024-08-22 at 22 26 59](https://github.com/user-attachments/assets/529f565b-2317-41d0-8e17-727b1093e204)

Si quieres investigar más sobre el uso de Codespaces acá hay una pequeña [guía de GitHub](https://docs.github.com/en/codespaces/getting-started/quickstart).

Saludos!
